### PR TITLE
Add SmartContent template filtering

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

Adds SmartContent template filtering for articles.

#### BC Breaks/Deprecations

Describe BC breaks/deprecations here. (remove this section if not needed)

#### To Do

- [ ] Add documentation page

### CODE REVIEW

File: Content/ArticleDataProvider.php [hunk 1]
```diff
@@ -140,8 +164,8 @@ public function resolveDataItems(
         $page = 1,
         $pageSize = null
     ) {
+        $filters['structureTypes'] = array_merge($filters['types'] ?? [], $this->getStructureTypesProperty($propertyParameter));
```
Comment (nnatter) [inline Content/ArticleDataProvider.php:167]: the mixup between `structureTypes` and `types` is confusing 😕  but i guess there is no better solution?

Comment (nnatter) [reply Content/ArticleDataProvider.php:167]: maybe we can at least add a comment that tries to explain what is what?

Comment (Prokyonn) [reply Content/ArticleDataProvider.php:167]: Yeah it is, we definitely should change this, but I don't think there is a better solution for the time being without changing too much

Comment (wachterjohannes) [reply Content/ArticleDataProvider.php:167]: @Prokyonn yeah please add a comment

**Review by wachterjohannes — APPROVED** [review]

**Pull request merged**